tests/test_cases/test_cocotb/test_cocotb.py

```diff
@@ -1251,5 +1251,31 @@ def test_assertion_is_failure(dut):
     assert False
 
 
+@cocotb.test(expect_error=True)
+def test_timeout_func_fail(dut):
+    @cocotb.coroutine
+    def fail():
+        yield Timer(10, 'ns')
+    yield cocotb.triggers.with_timeout(fail(), timeout_time=1, timeout_unit='ns')
+
+
+@cocotb.test(expect_error=False)
+def test_timeout_func_pass(dut):
+    @cocotb.coroutine
+    def success():
+        yield Timer(10, 'ns')
+    yield cocotb.triggers.with_timeout(success(), timeout_time=100, timeout_unit='ns')
+
+
+@cocotb.test(expect_error=True, timeout_time=1, timeout_unit='ns')
+def test_timeout_testdec_fail(dut):
+    yield Timer(10, 'ns')
+
+
+@cocotb.test(expect_error=False, timeout_time=100, timeout_unit='ns')
+def test_timeout_testdec_pass(dut):
+    yield Timer(10, 'ns')
```

Comment:
Can you add a test where the two timeouts are identical, and make a judgement call on whether you expect it to fail?

Reply:
I had considered this. This depends upon the scheduler's behavior doesn't it? If you yield a list, and two triggers fire simultaneously, the scheduler returns the first trigger in the list. If the scheduler does not guarantee what I think it does, and my tests are happy accidents, or if you expect the scheduler's behavior to change, the behavior of this function in that case will also change. I'd probably consider this case UB.

Reply:
> This depends upon the scheduler's behavior doesn't it?

Yes it does.

I'd still like to have a test for it though, so that we know if the behavior changed - I think tests for UB are still sensible, as long as they're labelled to indicate that it's ok to change the test if it fails.

Reply:
test was added